Signal/src/ViewControllers/SelectRecipientViewController.m

```diff
@@ -0,0 +1,553 @@
+{
+
+    [self updateCountryWithName:countryName callingCode:callingCode countryCode:countryCode];
+
+    [self textField:self.phoneNumberTextField shouldChangeCharactersInRange:NSMakeRange(0, 0) replacementString:@""];
+}
+
+#pragma mark - UITextFieldDelegate
+
+// TODO: This logic resides in both RegistrationViewController and here.
+//       We should refactor it out into a utility function.
+- (BOOL)textField:(UITextField *)textField
+    shouldChangeCharactersInRange:(NSRange)range
+                replacementString:(NSString *)insertionText
+{
+    [ViewControllerUtils phoneNumberTextField:textField
+                shouldChangeCharactersInRange:range
+                            replacementString:insertionText
+                                  countryCode:_callingCode];
+
+    [self updatePhoneNumberButtonEnabling];
+
+    return NO; // inform our caller that we took care of performing the change
+}
+
+- (BOOL)textFieldShouldReturn:(UITextField *)textField
+{
+    [textField resignFirstResponder];
+    [self tryToSelectPhoneNumber];
+    return NO;
+}
+
+#pragma mark - Table Contents
+
+- (void)updateTableContents
+{
+    OWSTableContents *contents = [OWSTableContents new];
+
+    __weak SelectRecipientViewController *weakSelf = self;
+    ContactsViewHelper *helper = self.contactsViewHelper;
+
+    OWSTableSection *phoneNumberSection = [OWSTableSection new];
+    phoneNumberSection.headerTitle = [self.delegate phoneNumberSectionTitle];
+    const CGFloat kCountryRowHeight = 50;
+    const CGFloat kPhoneNumberRowHeight = 50;
+    const CGFloat kButtonRowHeight = 60;
+    [phoneNumberSection addItem:[OWSTableItem itemWithCustomCellBlock:^{
+        SelectRecipientViewController *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return (UITableViewCell *)nil;
+        }
+
+        UITableViewCell *cell = [UITableViewCell new];
+
+        // Country Row
+        UIView *countryRow = [self createRowWithHeight:kCountryRowHeight previousRow:nil superview:cell.contentView];
+        [countryRow
+            addGestureRecognizer:[[OWSAnyTouchGestureRecognizer alloc] initWithTarget:self
+                                                                               action:@selector(countryRowTouched:)]];
+
+        UILabel *countryCodeLabel = self.countryCodeLabel;
+        [countryRow addSubview:countryCodeLabel];
+        [countryCodeLabel autoPinEdgeToSuperviewEdge:ALEdgeLeft withInset:20.f];
+        [countryCodeLabel autoVCenterInSuperview];
+
+        [countryRow addSubview:self.countryCodeButton];
+        [self.countryCodeButton autoPinEdgeToSuperviewEdge:ALEdgeRight withInset:20.f];
+        [self.countryCodeButton autoVCenterInSuperview];
+
+        // Phone Number Row
+        UIView *phoneNumberRow =
+            [self createRowWithHeight:kPhoneNumberRowHeight previousRow:countryRow superview:cell.contentView];
+        [phoneNumberRow addGestureRecognizer:[[OWSAnyTouchGestureRecognizer alloc]
```

Comment:
Okay, I've made this a tap.  We should discuss more generally where to use the "any touch" GR.  The idea is:

* "any touch" recognizes faster - as soon as the touch begins.
* "any touch" is way more permissive and accepts any number of touches / movement.  

I like to use it in places where there's no ambiguity about the user intent in the spirit of "be permissive about what you accept."